lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -4073,6 +4070,35 @@ ActorIsolation ActorIsolationRequest::evaluate(
       return ActorIsolation::forActorInstanceParameter(actor, *paramIdx);
   }
 
+  // Diagnose global state that is not either immutable plus Sendable or
+  // isolated to a global actor.
+  auto checkGlobalIsolation = [var = dyn_cast<VarDecl>(value)](
+                                  ActorIsolation isolation) {
+    if (var && var->getLoc() &&
+        var->getASTContext().LangOpts.hasFeature(Feature::GlobalConcurrency) &&
+        !isolation.isGlobalActor()) {
+      const bool isGlobalState =
+          var->isStatic() || var->getDeclContext()->isModuleScopeContext() ||
+          (var->getDeclContext()->isTypeContext() && !var->isInstanceMember());
+      auto *classDecl = var->getDeclContext()->getSelfClassDecl();
+      const bool isActorType = classDecl && classDecl->isActor();
```

Comment:
This should be using `isAnyActor` I think?

```

bool NominalTypeDecl::isAnyActor() const {
  return isActor() || isDistributedActor();
}
```

Please add a test using a `distributed actor` as well

Reply:
Well I don't think it would hurt to make it `isAnyActor`. Distributed actors won't reach this code path at this point anyway, since they can't be global, but if someday they can, then things will continue to work (and no explanatory comment required).

Reply:
Yeah maybe can use the isAnyActor right away, your call :) 

Thank you!